src/main/scala-2.12/org/apache/james/gatling/imap/scenari/ImapListMessagesBodyStructureScenario.scala

```diff
@@ -0,0 +1,32 @@
+package org.apache.james.gatling.imap.scenari
+
+import com.linagora.gatling.imap.PreDef._
+import com.linagora.gatling.imap.protocol.command.FetchAttributes.AttributeList
+import com.linagora.gatling.imap.protocol.command.MessageRange.From
+import com.linagora.gatling.imap.protocol.command.MessageRanges
+import io.gatling.core.Predef._
+import io.gatling.core.feeder.FeederBuilder
+import io.gatling.core.structure.ScenarioBuilder
+import org.apache.james.gatling.jmap.MailboxName
+
+import scala.concurrent.duration._
+
+class ImapListMessagesBodyStructureScenario {
+  val random = scala.util.Random
+
+  val listMessagesBodyStructure = exec(imap("select").select("${mailboxName}").check(ok))
+    .exec(imap("fetch").fetch(MessageRanges(From(1)), AttributeList("BODYSTRUCTURE")).check(ok))
+
+  def generate(feeder: FeederBuilder, mailboxes: Iterable[MailboxName], mailboxesListed: Int): ScenarioBuilder =
+    scenario("ImapListMessagesBodyStructureScenario")
+      .feed(feeder)
+      .pause(1.second)
+      .exec(imap("Connect").connect()).exitHereIfFailed
+      .exec(imap("login").login("${username}", "${password}").check(ok))
+      .exec{session =>
+        session.set("mailboxes", random.shuffle(mailboxes.map(_.name)).take(mailboxesListed))
+      }
+      .exec(foreach("${mailboxes}", "mailboxName") {
```

Comment:
You may create a group for the part you want to assert the execution time.
Then in the simulation you will be able to add assertion for the executime of this group.
You can see example of this in the JMAP scenario and simulations on the mail corpus